test/metrics/cc/test_all_types.py

```diff
@@ -1,37 +1,189 @@
 # SPDX-FileCopyrightText: Copyright (c) 2019-2025 Aibolit
 # SPDX-License-Identifier: MIT
 
-import unittest
-
-
-class JavaTestCase(unittest.TestCase):
-    @classmethod
-    def setUpClass(cls):
-        super(JavaTestCase, cls).setUpClass()
-
-    def runAnalysis(self):
-        super(JavaTestCase, self).setUp()
-        from aibolit.metrics.cc.main import CCMetric
-
-        file = 'test/metrics/cc/Complicated.java'
-        metric = CCMetric(file)
-        res = metric.value(True)
-        self.assertEqual(res['data'][0]['complexity'], 12)
-        self.assertEqual(res['data'][0]['file'], file)
-
-        with self.assertRaises(Exception) as context:
-            file = 'test/metrics/cc/ooo.java'
-            metric = CCMetric(file)
-            res = metric.value(True)
-        self.assertTrue('PMDException' in str(context.exception))
-
-        with self.assertRaises(Exception) as context:
-            file = 'test/metrics/cc/ooo1.java'
-            metric = CCMetric(file)
-            res = metric.value(True)
-        self.assertTrue('File test/metrics/cc/ooo1.java does not exist' == str(context.exception))
-
-        file = 'test/metrics/cc/OtherClass.java'
-        metric = CCMetric(file)
-        res = metric.value(True)
-        self.assertEqual(res['data'][0]['complexity'], 3)
+from unittest import TestCase
+from pathlib import Path
+from textwrap import dedent
+
+from aibolit.metrics.cc.main import CCMetric
+from aibolit.ast_framework import AST
+from aibolit.utils.ast_builder import build_ast, build_ast_from_string
+
+
+class CCTestCase(TestCase):
+    current_directory = Path(__file__).absolute().parent
+
+    @staticmethod
+    def _get_ast(filename: str) -> AST:
+        path = Path(__file__).absolute().parent / filename
+        return AST.build_from_javalang(build_ast(str(path)))
+
+    def test_complicated_example(self):
+        ast = self._get_ast('Complicated.java')
+        metric = CCMetric()
+        res = metric.value(ast)
+        self.assertEqual(res, 12)
+
+    def test_other_class_example(self):
+        ast = self._get_ast('OtherClass.java')
+        metric = CCMetric()
+        res = metric.value(ast)
+        self.assertEqual(res, 3)
+
+    def test_empty_method(self):
+        content = dedent(
+            '''\
+            class Dummy {
+              private void empty() {}
+            }
+            '''
+        ).strip()
+        self.assertEqual(self._cc_metric_for(content), 1)
+
+    def test_simple_if_statement(self):
+        content = dedent(
+            '''\
+            class Dummy {
+              private void test(boolean condition) {
+                if (condition) {
+                  System.out.println("true");
+                }
+              }
+            }
+            '''
+        ).strip()
+        self.assertEqual(self._cc_metric_for(content), 2)
+
+    def test_if_else_statement(self):
```

Comment:
Thank you for adding these tests!

I suppose that we can think of some additional tests with nested if/else statements to make the test suite even more solid.